llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -15167,8 +15169,9 @@ bool RISCV::CC_RISCV(const DataLayout &DL, RISCVABI::ABI ABI, unsigned ValNo,
   MVT XLenVT = XLen == 32 ? MVT::i32 : MVT::i64;
   MVT CLenVT = Subtarget.hasCheri() ? Subtarget.typeForCapabilities()
                                     : MVT();
-  MVT PtrVT = DL.isFatPointer(DL.getAllocaAddrSpace()) ? CLenVT : XLenVT;
-  bool IsPureCapVarArgs = !IsFixed && RISCVABI::isCheriPureCapABI(ABI);
+  bool IsPureCap = RISCVABI::isCheriPureCapABI(ABI);
+  MVT PtrVT = IsPureCap ? CLenVT : XLenVT;
+  bool IsPureCapVarArgs = !IsFixed && IsPureCap;
```

Comment:
I wrote the code like the LHS because PtrVT is a stack pointer and so should be tied to the alloca AS. In practice the RHS is entirely equivalent because the alloca AS is a capability if and only if a purecap ABI is in use, but I was trying to be as general as possible, using fine-grained ASes where appropriate.

Reply:
Also the commit message you used was:
> [CHERI] Use ABI to deduce purecap instead of default addrspace 
which isn't appropriate here, because this is the alloca AS, not the default globals AS.

Reply:
Do you mind reverting this part? I thought it didn't matter since `alloca AS == 200` is always equivalent to purecap and we were checking the ABI later again.

Reply:
Thanks, I understand the difference now. Please revert if needed or I can push a patch which just resets this particular hunk?